coordinator/txmanager.go

```diff
@@ -129,29 +130,36 @@ func (t *TxManager) syncSCVars(vars common.SCVariablesPtr) {
 
 // NewAuth generates a new auth object for an ethereum transaction
 func (t *TxManager) NewAuth(ctx context.Context, batchInfo *BatchInfo) (*bind.TransactOpts, error) {
-	//First we try getting the gas price from etherscan. If it fails we get the gas price from the ethereum node.
-	var gasPrice *big.Int
+	// First we try getting the gas price from etherscan. If it fails we get the gas price from the ethereum node.
+
+	// If gas price is higher than 2000, probably we are going to get the gasLimit exceed error
+	const maxGasPrice = 2000
+	// If gas price is to low, probably there is an error or at Etherscan or Ethereum node.
+	// So setting a minimum value
+	const minGasPrice = 5
+
+	gasPrice := big.NewInt(minGasPrice)
+	eGasPrice := big.NewInt(minGasPrice)
+
 	if t.etherscanService != nil {
 		etherscanGasPrice, err := t.etherscanService.GetGasPrice(ctx)
 		if err != nil {
-			log.Warn("Error getting the gas price from etherscan. Trying another method. Error: ", err.Error())
+			log.Warn("[Etherscan gas price service] Error getting the gas price from etherscan. Trying another method. Error: ", err.Error())
 			var er error
 			gasPrice, er = t.ethClient.EthSuggestGasPrice(ctx)
 			if er != nil {
 				return nil, tracerr.Wrap(er)
 			}
 		} else {
-			eGasPrice := new(big.Int)
-			eGasPrice, ok := eGasPrice.SetString(etherscanGasPrice.ProposeGasPrice, 10)
+			var ok bool
+			eGasPrice, ok = eGasPrice.SetString(etherscanGasPrice.ProposeGasPrice, 10)
```

Comment:
As sometimes the gas price is not enough, maybe we should use FastGasPrice instead of ProposeGasPrice. This option is expensive but the tx should always be mined. {"LastBlock":"12529900","SafeGasPrice":"19","ProposeGasPrice":"35","FastGasPrice":"42"}